### PR TITLE
Implement StageChannel and related methods

### PR DESCRIPTION
**Pull request by NCPlayz**
## Summary

Implements the new features as part of channel type ``13``, guild stage voice channels.
Reviews welcome.

Additions:

- `ChannelType.stage_voice`
- `StageChannel`
- `StageChannelConverter`
- `Guild.stage_channels`
- `CategoryChannel.stage_channels`
- `Guild.create_stage_channel`
- `CategoryChannel.create_stage_channel`
- `Member.request_to_speak`
- `Member.edit(suppress=True | False)`
- `VoiceState.suppress`
- `VoiceState.requested_to_speak_at`
- `Permissions.stage_moderator()`

## Tests
<details><summary>
Requesting to speak
</summary>
<p>

```py
await guild.me.request_to_speak()
```

proof that it works

</p>
</details>

<details><summary>
Requesting others to speak
</summary>
<p>

```py
await connected_member.request_to_speak()
# or
await connected_member.edit(suppress=False)
```

proof that it works

</p>
</details>

<details><summary>
Creating new stage channels
</summary>
<p>

```py
>>> await guild.create_stage_channel(name='test')
# or
>>> await category.create_stage_channel(name='test')
<StageChannel id=123456789 name='test' topic=None rtc_region=None position=1 bitrate=40000 user_limit=1000 category_id=123456789>
```

proof that it works

</p>
</details>

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/member.py [hunk 1]
```diff
@@ -59,15 +61,32 @@ class VoiceState:
...
+        anymore or have been accepted to speak.
+
+        Only applicable to stage channels.
+
+        .. versionadded:: 1.7
+
     afk: :class:`bool`
         Indicates if the user is currently in the AFK channel in the guild.
-    channel: Optional[:class:`VoiceChannel`]
+    channel: Optional[:class:`VoiceChannel`, :class:`StageChannel`]
```
Comment (Rapptz) [inline discord/member.py:82]:
```suggestion
    channel: Optional[Union[:class:`VoiceChannel`, :class:`StageChannel`]]
```

**Review by Rapptz — CHANGES_REQUESTED** [review]
I haven't tested it but the code seems mostly ok

**Comment by Tari-dev** [comment]
ummmm... GuildVocalChannel sounds better than VocalGuildChannel.

**Comment by diceroll123** [comment]
> ummmm... GuildVocalChannel sounds better than VocalGuildChannel.

*Sounds* better, but it is a GuildChannel at its base.

**Comment by Rapptz** [comment]
It's not exposed publicly so it's name does not matter.

**Comment by diceroll123** [comment]
Gotta update `Permissions.all()` to include 1 << 32

File: discord/channel.py [hunk 2]
```diff
@@ -728,6 +732,130 @@ async def edit(self, *, reason=None, **options):
...
+            ('category_id', self.category_id)
+        ]
+        return '<%s %s>' % (self.__class__.__name__, ' '.join('%s=%r' % t for t in attrs))
+
+    def _update(self, guild, data):
+        super()._update(guild, data)
+        self.topic = data.get('topic')
+
+    @property
+    def requests_to_speak(self):
```
Comment (Rapptz) [inline discord/channel.py:801]:
```suggestion
    def requesting_to_speak(self):
```

**Review by Rapptz — APPROVED** [review]
Looks good. I'll manually commit the one suggestion I have.

**Pull request merged**